game/hud/src/services/session/layout.ts

```diff
@@ -109,18 +121,80 @@ const initialState = {
   widgets: defaultWidgets(),
 }
 
-let getInitialState = () => {
-  var storedState = JSON.parse(localStorage.getItem(localStorageKey)) as LayoutState;
+interface Size {
+  width: number;
+  height: number;
+}
+
+const px2pcnt = (pos: Position, screen: Size) : Position => {
+  // converts pixel co-ordinates into % of display size
+  return {
+    x: pos.x / screen.width,
+    y: pos.y / screen.height,
+    width: pos.width,           // don't scale width/height
+    height: pos.height,
+    scale: 1
+  };
+};
+
+const pcnt2px = (pos: Position, screen: Size) : Position => {
+  // converts % co-ordinates into pixels for current display
+  return {
+    x: screen.width * pos.x,
+    y: screen.height * pos.y,
+    width: pos.width,           // don't scale width/height
+    height: pos.height,
+    scale: 1
+  }
+};
+
+const forceOnScreen = (pos: Position, screen: Size) : Position => {
+  if (pos.x < 0) pos.x = 0;
+  if (pos.y < 0) pos.y = 0;
+  if (pos.x + pos.width > screen.width) pos.x = screen.width - pos.width;
+  if (pos.y + pos.height > screen.height) pos.y = screen.height - pos.height;
+  if (pos.x < 0) { pos.x = 0; pos.width = screen.width; }
+  if (pos.y < 0) { pos.y = 0; pos.height = screen.height; }
+  return pos;
+};
+
+const getInitialState = (): any => {
+  const storedState: LayoutState = loadState();
```

Comment:
8.1 is talking about the use of arrow functions not the definition of functions via arrows. It does not allow it.

However JB has said that the existing conventions of of the codebase are king so it is allowed